dashboard/src/components/Tabs/tabsUtils.ts

```diff
@@ -0,0 +1,34 @@
+import { useMemo } from 'react';
+
+export type RecordDiffBooleanType = Record<string, Record<string, boolean>>;
+export type RecordDiffNumberType = Record<string, number>;
+
+export type RecordDiffType = RecordDiffNumberType | RecordDiffBooleanType;
+
+export const useDiffFilterParams = <
+  TFilterObjectsKeys extends string,
+  TFilter extends Record<TFilterObjectsKeys, Record<string, boolean>>,
```

Comment:
nit: rename this to TFilterGeneric or something similar in order to not confuse with specific TFilter types from treeDetails or hardwareDetails